Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+# Java Basics
+
+## Core Concepts
+
+### The Java Virtual Machine (JVM) 
+
+Java belongs to a family of languages called **Compiled Languages** (https://en.wikipedia.org/wiki/Compiled_language). Any code written in such a language needs to be converted (compiled) to an intermediate form that can then be understood by the host platform (the OS/platform in which the code runs). For Java, this intermediate form is called **Bytecode** which is then interpreted by a runtime called a Java Virtual Machine (JVM). Think of JVM (https://docs.oracle.com/javase/specs/jvms/se7/html/) as a piece of software that does the hard work of running your Java code. It takes care of memory allocation, thread management, garbage collection and so much more. Apart from Java, it also supports (read: able to run) code written in languages such as Groovy, Scala etc.
+
+In Java, code is written and saved as `.java` files. The files are then run through a compiler (javac) to convert them to Bytecode `.class` files. Standalone Java programs are then run using the `java` command. More about this later. 
```

Comment:
> The files are then run through a compiler (javac)

Not exactly a sentence that should appear in a technical writing. A file is a collection of strings, stored in memory. 

The word "run" is associated with something that can be executed.

Phrase it better; with perhaps something like

> The Java compiler operates on the file, to generate equivalent Bytecode, stored in `.class` files.